projects/driver_display/src/main.c

```diff
@@ -0,0 +1,53 @@
+#include <stddef.h>
+#include <stdio.h>
+
+#include "interrupt.h"
+
+#include "driver_display_brightness.h"
+
+int main(void) {
+  // Init everything to be used
+  interrupt_init();
+  soft_timer_init();
+  gpio_init();
+  adc_init(ADC_MODE_CONTINUOUS);
+
+  // Populate data for brightness module (temporarily here before calibration is implemented)
+  DriverDisplayBrightnessScreenData screen_data[] = {
```

Comment:
This is not easy to use. You should have a settings struct (or if there are only a few options, just pass in parameters) that has the bare minimum required to initialize the module. I shouldn't have to set up `GPIOSettings` for you.